### ISSUE TITLE
Bug in `MLJModelInterface.update`

### ISSUE THREAD

**Issue by ablaom** [Evovest/EvoTrees.jl#92]
Ran into a bug when trying to tune a EvoTreeRegressor in MLJ. Isolated it to this:

```julia
using MLJ
using EvoTrees
using MLJModelInterface
const MMI = MLJModelInterface

X, y = @load_boston
model = (@load EvoTreeRegressor)()
data = MMI.reformat(model, X, y)
f, c, r = MMI.fit(model, 2, data...);
model.λ = 0.1

julia> MMI.update(model, 2, f, c, data...);
ERROR: ArgumentError: Function `matrix` only supports AbstractMatrix or containers implementing the Tables interface.
Stacktrace:
 [1] matrix(::FullInterface, ::Val{:other}, X::NamedTuple{(:matrix, :names), Tuple{Matrix{Float64}, Vector{Symbol}}}; kw::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ MLJModelInterface ~/.julia/packages/MLJModelInterface/tegnW/src/data_utils.jl:32
 [2] matrix(::FullInterface, ::Val{:other}, X::NamedTuple{(:matrix, :names), Tuple{Matrix{Float64}, Vector{Symbol}}})
   @ MLJModelInterface ~/.julia/packages/MLJModelInterface/tegnW/src/data_utils.jl:32
 [3] matrix(X::NamedTuple{(:matrix, :names), Tuple{Matrix{Float64}, Vector{Symbol}}}; kw::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ MLJModelInterface ~/.julia/packages/MLJModelInterface/tegnW/src/data_utils.jl:27
 [4] matrix
   @ ~/.julia/packages/MLJModelInterface/tegnW/src/data_utils.jl:27 [inlined]
 [5] reformat(#unused#::EvoTrees.EvoTreeRegressor{Float64, EvoTrees.Linear, Int64}, X::NamedTuple{(:matrix, :names), Tuple{Matrix{Float64}, Vector{Symbol}}}, y::SubArray{Float64, 1, Matrix{Float64}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true})
   @ EvoTrees ~/.julia/packages/EvoTrees/oHLKA/src/MLJ.jl:24
 [6] update(model::EvoTrees.EvoTreeRegressor{Float64, EvoTrees.Linear, Int64}, verbosity::Int64, fitresult::EvoTrees.GBTree{1, Float64, Int64}, cache::NamedTuple{(:params, :X, :Y_cpu, :pred_cpu, :𝑖_, :𝑗_, :𝑖, :𝑗, :δ, :δ², :𝑤, :edges, :X_bin, :train_nodes, :splits, :hist_δ, :hist_δ², :hist_𝑤), Tuple{EvoTrees.EvoTreeRegressor{Float64, EvoTrees.Linear, Int64}, Matrix{Float64}, Vector{Float64}, Vector{StaticArrays.SVector{1, Float64}}, Vector{Int64}, Vector{Int64}, Vector{Int64}, Vector{Int64}, Vector{StaticArrays.SVector{1, Float64}}, Vector{StaticArrays.SVector{1, Float64}}, Vector{StaticArrays.SVector{1, Float64}}, Vector{Vector{Float64}}, Matrix{UInt8}, Vector{EvoTrees.TrainNode{1, Float64, Int64}}, Vector{EvoTrees.SplitInfo{1, Float64, Int64}}, Vector{Matrix{StaticArrays.SVector{1, Float64}}}, Vector{Matrix{StaticArrays.SVector{1, Float64}}}, Vector{Matrix{StaticArrays.SVector{1, Float64}}}}}, A::NamedTuple{(:matrix, :names), Tuple{Matrix{Float64}, Vector{Symbol}}}, y::SubArray{Float64, 1, Matrix{Float64}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true})
   @ EvoTrees ~/.julia/packages/EvoTrees/oHLKA/src/MLJ.jl:39
 [7] top-level scope
   @ REPL[183]:1
```

Happy to look into this.

**Comment by ablaom** [comment]
I found a redundant `reformat` call in `update`.  Will make a PR.

**Comment by jeremiedb** [comment]
Fixed with #93
